contracts/protocol/tokenization/NTokenApeStaking.sol

```diff
@@ -81,19 +96,88 @@ abstract contract NTokenApeStaking is NToken {
         address receiverOfUnderlying,
         uint256[] calldata tokenIds
     ) external virtual override onlyPool nonReentrant returns (uint64, uint64) {
-        ApeStakingLogic.executeAutoWithdraw(
-            _ERC721Data.owners,
-            _apeCoinStaking,
-            POOL_ID(),
-            tokenIds,
-            from
+        require(
+            !ApeStakingLogic.isApeStakingPositionExisted(
+                POOL_ID(),
+                _apeCoinStaking,
+                tokenIds
+            ),
+            Errors.APE_STAKING_POSITION_EXISTED
         );
 
         return _burn(from, receiverOfUnderlying, tokenIds);
     }
 
-    function POOL_ID() internal virtual returns (uint256) {
+    function startAuction(uint256 tokenId)
+        public
+        virtual
+        override
+        onlyPool
+        nonReentrant
+    {
+        uint256[] memory tokenIds = new uint256[](1);
+        tokenIds[0] = tokenId;
+        require(
+            !ApeStakingLogic.isApeStakingPositionExisted(
+                POOL_ID(),
+                _apeCoinStaking,
+                tokenIds
+            ),
+            Errors.APE_STAKING_POSITION_EXISTED
+        );
+        super.startAuction(tokenId);
+    }
+
+    function POOL_ID() internal pure virtual returns (uint256) {
         // should be overridden
         return 0;
     }
+
+    function initializeStakingData() internal {
+        ApeStakingLogic.APEStakingParameter
+            storage dataStorage = apeStakingDataStorage();
+        ApeStakingLogic.executeSetUnstakeApeHFLimit(dataStorage, 1.1e18);
+        ApeStakingLogic.executeSetUnstakeApeIncentive(dataStorage, 30);
+    }
+
+    function setUnstakeApeHFLimit(uint256 hfLimit) external onlyPoolAdmin {
+        ApeStakingLogic.executeSetUnstakeApeHFLimit(
+            apeStakingDataStorage(),
+            hfLimit
+        );
+    }
+
+    function setUnstakeApeIncentive(uint256 incentive) external onlyPoolAdmin {
+        ApeStakingLogic.executeSetUnstakeApeIncentive(
+            apeStakingDataStorage(),
+            incentive
+        );
+    }
+
+    function apeStakingDataStorage()
+        internal
+        pure
+        returns (ApeStakingLogic.APEStakingParameter storage rgs)
+    {
+        bytes32 position = APE_STAKING_DATA_STORAGE_POSITION;
+        assembly {
+            rgs.slot := position
+        }
+    }
+
+    function unstakePositionAndRepay(uint256 tokenId, address unstaker)
```

Comment:
I think we can call it `liquidateSApe` because it's similar to liquidation and the result is that users will  lose part of staked ape